go/vt/vtctl/reparentutil/util.go

```diff
@@ -97,10 +97,10 @@ func ChooseNewPrimary(
 		tb := tablet.Tablet
 		errorGroup.Go(func() error {
 			// find and store the positions for the tablet
-			pos, err := findPositionForTablet(groupCtx, tb, logger, tmc, waitReplicasTimeout)
+			pos, replLag, err := findPositionAndLagForTablet(groupCtx, tb, logger, tmc, waitReplicasTimeout)
 			mu.Lock()
 			defer mu.Unlock()
-			if err == nil {
+			if err == nil && waitReplicasTimeout >= replLag {
```

Comment:
> That can be very flappy. A value of 15 seconds does not mean that it won't be 0 in a few milliseconds

Yes, this is understood. The feature request is still that callers should be able to specify a threshold for allowed replication lag that we check _before_ starting the failover.

@GuptaManan100 we'll need to make this a flag so that it can be specified separately from `waitReplicasTimeout`.

Reply:
> Which means that we need another flag :(

or we make it part of a "durability" policy

Reply:
I don't think we should put it into the `durability policy` because it would require us hardcoding some value and doubling all our existing durability policies that we offer out of the box to support both use-cases.

Reply:
I think a flag is in order? I'm going to make the changes and we can choose. If we don't want the flag, I'll remove it.